src/components/content/order/create/useAutoFillDeployVariableQuery.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { AutoFill, CloudResourcesService, DeployRequest } from '../../../../xpanse-api/generated';
+import { useQuery } from '@tanstack/react-query';
+
+export default function useAutoFillDeployVariableQuery(
+    csp: DeployRequest.csp,
+    region: string,
+    kind: AutoFill.deployResourceKind | undefined
+) {
+    return useQuery({
+        queryKey: ['autoFillDeployVariable', csp, region, kind],
```

Comment:
first value in queryKey can be the method name "getExistingResourcesOfType" instead of `autoFillDeployVariable`